saleor/graphql/warehouse/mutations.py

```diff
@@ -165,3 +166,15 @@ class Meta:
 
     class Arguments:
         id = graphene.ID(description="ID of a warehouse to delete.", required=True)
+
+    @classmethod
+    def perform_mutation(cls, _root, info, **data):
+        manager = info.context.plugins
+        node_id = data.get("id")
+        model_type = cls.get_type_for_model()
+        instance = cls.get_node_or_error(info, node_id, only_type=model_type)
+        stocks = (stock for stock in instance.stock_set.only("product_variant"))
+        result = super(WarehouseDelete, cls).perform_mutation(_root, info, **data)
+        for stock in stocks:
+            trigger_out_of_stock_webhook(stock, manager)
```

Comment:
We should also check that we receive a proper payload in delete mutation. If you generate payload after you remove stock. `stock.id` for this object will be `None`.

Reply:
1. Body of `trigger_out_of_stock_webhook` function already contain `transaction.on_commit`
2. I've created a list of stocks to trigger before any mutation. I have a suitable test case for that.

Reply:
1. Tests are executed inside a transaction. This mutation doesn't use transactions. 
2. This test checks only `trigger_out_of_stock_webhook` is called. If you generate payload from a removed object you receive `object_graphql_id=base64("TypeName: None")` instead of `object_graphql_id=base64("TypeName: <Object.pk>")`